packages/node/src/handlers.ts

```diff
@@ -112,6 +112,17 @@ function extractRequestData(req: { [key: string]: any }): { [key: string]: strin
     url: absoluteUrl,
   };
 
+  const attributes = Array.isArray(keys) ? keys : [];
+
+  if (attributes.length) {
```

Comment:
Using slightly different logic here than the `extractUserData` function, as you are already running some custom logic to set/manipulate keys on the request interface.
We build the request interface as before, this addition then removes any of the properties not specified in the optional array (if it exists).

This behaviour is slightly different to the `extractUserData` fn as it allows users to request any of the properties from the request interface rather than any properties of the `req` argument itself.

Reply:
If that's a bit too ambiguous we could do something like below:
 1. Set the requested properties
 2. Run the rest of the current logic to overwrite the values
 3. Merge the requested properties with the default values
 4. Delete any keys that don't match the requested properties from the array.

```
  const request: { [key: string]: string } = {};
  const attributes = Array.isArray(keys) ? keys : [];

  attributes.forEach(key => {
    if ({}.hasOwnProperty.call(req, key)) {
      request[key] = (req as { [key: string]: string })[key];
    }
  });

  ----
  // set values
  const headers = ...
  const method = ...
  ----

  // request interface
  const requestInterface: {
    [key: string]: any;
  } = {
    ...request
    cookies,
    data,
    headers,
    method,
    query_string: query,
    url: absoluteUrl,
  };

 ----
 // Delete any of the default keys not specified in the array
  if (attributes.length) {
    Object.keys(requestInterface).forEach(key => {
      /** Remove any of the unspecified keys in the options from the request interface */
      if (!attributes.includes(key)) {
        delete requestInterface[key];
      }
    });
  }

 return requestInterface


```